lib/src/util/dart_type_utilities.dart

```diff
@@ -367,7 +367,7 @@ class DartTypeUtilities {
         // Otherwise, they might be related.
         return false;
       } else {
-        return leftElement.supertype.isObject ||
+        return leftElement.supertype?.isObject == true ||
```

Comment:
Test?

Reply:
Still trying to distill a repro.  Maybe @bwilkerson has ideas?

Reply:
Sorry, no. As far as I can remember the supertype ought to always be non-null, so I suspect this is really hiding a bug in analyzer.

Reply:
According to https://pub.dev/documentation/analyzer/latest/dart_element_element/ClassElement/supertype.html:

> Return the superclass of this class, or `null` if either the class represents the class 'Object' or if the class represents a mixin declaration. All other classes will have a non-`null` superclass.